game/src/edit/traffic_signals/edits.rs

```diff
@@ -62,32 +57,34 @@ impl ChangeDuration {
                 })
                 .into_widget(ctx)
                 .named("timing type"),
-                "How much additional time can this stage last?".text_widget(ctx),
                 Widget::row(vec![
-                    "Seconds:".text_widget(ctx).centered_vert(),
+                    "How much additional time can this stage last?"
+                        .text_widget(ctx)
+                        .centered_vert(),
                     Spinner::widget(
                         ctx,
                         "additional",
-                        (0, 300),
+                        (Duration::ZERO, Duration::minutes(5)),
                         match signal.stages[idx].stage_type {
-                            StageType::Fixed(_) => 0,
-                            StageType::Variable(_, _, additional) => {
-                                additional.inner_seconds() as isize
-                            }
+                            StageType::Fixed(_) => Duration::ZERO,
+                            StageType::Variable(_, _, additional) => additional,
                         },
+                        Duration::seconds(1.0),
                     ),
                 ]),
-                "How long with no demand before the stage ends?".text_widget(ctx),
                 Widget::row(vec![
-                    "Seconds:".text_widget(ctx).centered_vert(),
```

Comment:
I altered the layout a bit, since it's redundant now to label as "seconds." Yuwen has a nicer design for this panel to make the fixed/variable timing more clear; I'll try to get to that soon too.

Reply:
looks good!